api/interestr/api/serializers.py

```diff
@@ -34,13 +34,64 @@ class Meta:
 
 class DataTemplateSimpleSerializer(serializers.ModelSerializer):
 
+    VALID_TYPES = [
+        'checkbox',
+        'multisel',
+        'textarea',
+        'text',
+        'number',
+        'date',
+        'email',
+        'url',
+        'tel'
+    ]
+
     class Meta:
         model = core_models.DataTemplate
         fields = ('id', 'name', 'group', 'user',
                   'created', 'updated', 'fields')
 
+    def validate_fields(self, value):
+        """
+        Check that fields value is in correct format.
+        """
+        
+        errors = []
+
+        legends = []
+
+        i = 0
+        for field in value:
```

Comment:
No errors are caught in the case of value (fields) being empty. A template may not have its fields completely empty. It currently does on the web front-end, but it shouldn't really.

Reply:
True. Good point.